src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SecurityTokenHandlerCollection.cs

```diff
@@ -0,0 +1,686 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Security.Claims;
+using System.Xml;
+using CoreWCF.IdentityModel.Selectors;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Defines a collection of SecurityTokenHandlers.
+    /// </summary>
+    public class SecurityTokenHandlerCollection : Collection<SecurityTokenHandler>
+    {
+        internal static int s_defaultHandlerCollectionCount = 8;
+        internal static SecurityTokenHandlerCollection s_defaultSecurityTokenHandlerCollection = null; //TODO if better way
+        private readonly Dictionary<string, SecurityTokenHandler> _handlersByIdentifier = new Dictionary<string, SecurityTokenHandler>();
```

Comment:
You can't cache the default set as they might get modified.

Reply:
https://github.com/CoreWCF/CoreWCF/issues/585 => Pv2